rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,306 +150,167 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
-impl Message for UnListen {
-    type Result = ();
-}
-
-impl Handler<UnListen> for TunnelManager {
-    type Result = ();
-
-    fn handle(&mut self, un_listen: UnListen, _: &mut Context<Self>) -> Self::Result {
-        self.listen_interfaces.remove(&un_listen.0);
-        SETTING.get_network_mut().peer_interfaces = self.listen_interfaces.clone();
-    }
-}
-
-pub struct GetListen;
-impl Message for GetListen {
-    type Result = Result<HashSet<String>, Error>;
-}
-
-impl Handler<GetListen> for TunnelManager {
-    type Result = Result<HashSet<String>, Error>;
-    fn handle(&mut self, _: GetListen, _: &mut Context<Self>) -> Self::Result {
-        Ok(self.listen_interfaces.clone())
-    }
-}
-
 pub struct GetNeighbors;
 impl Message for GetNeighbors {
     type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
 }
 
 impl Handler<GetNeighbors> for TunnelManager {
-    type Result = ResponseFuture<Vec<(LocalIdentity, String, IpAddr)>, Error>;
+    type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
 
     fn handle(&mut self, _: GetNeighbors, _: &mut Context<Self>) -> Self::Result {
-        self.get_neighbors()
-    }
-}
-
-pub struct GetLocalIdentity {
-    pub from: IpAddr,
-}
-impl Message for GetLocalIdentity {
-    type Result = LocalIdentity;
-}
-
-impl Handler<GetLocalIdentity> for TunnelManager {
-    type Result = MessageResult<GetLocalIdentity>;
-
-    fn handle(&mut self, their_id: GetLocalIdentity, _: &mut Context<Self>) -> Self::Result {
-        MessageResult(self.get_local_identity(their_id.from))
+        let mut res = Vec::new();
+        for tunnel in self.tunnels.iter() {
+            res.push((tunnel.localid.clone(), tunnel.iface_name.clone(), tunnel.ip));
+        }
+        Ok(res)
     }
 }
 
-pub struct OpenTunnel(pub LocalIdentity, pub IpAddr);
+pub struct PeersToContact(pub Vec<Peer>);
 
-impl Message for OpenTunnel {
+impl Message for PeersToContact {
     type Result = ();
 }
 
-impl Handler<OpenTunnel> for TunnelManager {
+impl Handler<PeersToContact> for TunnelManager {
     type Result = ();
-
-    fn handle(&mut self, their_id: OpenTunnel, _: &mut Context<Self>) -> Self::Result {
-        self.open_tunnel(their_id.0, their_id.1).unwrap();
-        ()
+    fn handle(&mut self, peers: PeersToContact, _ctx: &mut Context<Self>) -> Self::Result {
+        for peer in peers.0.iter() {
+            let res = self.neighbor_inquiry(peer.clone());
+            if res.is_err() {
+                warn!("Neighbor inqury for {:?} failed!", peer);
+            }
+        }
+        for manual_peer in SETTING.get_network().manual_peers.iter() {
+            let ip = manual_peer.parse::<IpAddr>();
+            let port = SETTING.get_network().rita_hello_port;
+            if ip.is_ok() {
+                let ip = ip.unwrap();
+                let socket = SocketAddr::new(ip, port);
+                let man_peer = Peer {
+                    linklocal_ip: ip, // not actually linklocal this time
```

Comment:
maybe contact_ip is a better name? or maybe the LLIP is actually redundant when we have contact_socket